core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -1687,6 +1714,15 @@ class KafkaConfig(val props: java.util.Map[_, _], doLog: Boolean, dynamicConfigO
   val deleteTopicEnable = getBoolean(KafkaConfig.DeleteTopicEnableProp)
   def compressionType = getString(KafkaConfig.CompressionTypeProp)
 
+  /** ********* Raft Quorum Configuration *********/
+  val quorumVoters = getString(KafkaConfig.QuorumVotersProp)
```

Comment:
Can we add a check somewhere that `quorum.voters` is required if `process.roles` is non-empty?

Reply:
Spoke offline. The ClusterTestKit on the `kip-500` branch currently requires the `quorum.voters` to be empty at startup and to be set at a later point-in-time (once the controller starts and has chosen a free port). We don't want testing to dictate the norm here. So, to allow for that one alternative is to add a `skipVoterValidation` flag in a later PR. For now, I'll move the validation from RaftConfig to KafkaConfig and skip the validation if the voter string is empty

Reply:
Sounds good. Since this is an internal config at the moment, we have some time to consider a better long-term solution. My feeling is that we need a separate path for constructing the dependencies we need, but I guess we can cross that bridge when we get there.